### PR TITLE
chore: implement OHLC data digest system

### PR DESCRIPTION
**Pull request by MicBun**
Implement a comprehensive OHLC (Open, High, Low, Close) data digest system that enables efficient storage and retrieval of time-series financial data for both cryptocurrency and traditional stock markets.

- **Data Compression**: Reduces storage requirements by up to 75% by preserving only essential OHLC records per trading day
- **Analytics Performance**: Enables fast OHLC queries for financial analysis, charting, and reporting
- **Market Coverage**: Supports both 24/7 crypto markets and session-based stock markets with proper day boundary handling
- **Data Integrity**: Maintains exact OHLC values through deterministic tie-breaking rules

**Schema Changes:**
- Added `primitive_event_type` table with bitwise OHLC type flags (1=OPEN, 2=HIGH, 4=LOW, 8=CLOSE)
- Added `pending_prune_days` queue table for digest scheduling (removed unnecessary status column)
- Added `digest_config` table for system configuration

**Core Actions:**
- `digest_daily`: Processes single day data, preserves OHLC records, deletes redundant data
- `auto_digest`: Batch processes multiple days with configurable batch size
- `get_daily_ohlc`: Retrieves OHLC data from both raw and digested sources

**Key Features:**
- **Proper OHLC Ordering**: Follows conventional financial data standards (OHLC positional order)
- **Deterministic Logic**: OPEN=earliest time, HIGH=max value, LOW=min value, CLOSE=latest time
- **Combination Flags**: Bitwise arithmetic handles overlapping OHLC values (e.g., OPEN+LOW=5)
- **Leader Authorization**: Prepared for distributed consensus (currently disabled for testing)
- **Day Boundaries**: Accurate 24-hour periods using unix timestamps with 86400-second ranges

- **Type Flag Validation**: Verifies correct bitwise flags are assigned to OHLC records
- **Combination Scenarios**: Tests overlapping OHLC values with proper flag arithmetic
- **Data Integrity**: Confirms identical OHLC results from both raw and digested data
- **Edge Cases**: Handles single records, duplicate values, and missing data scenarios

- `internal/migrations/019-digest-schema.sql` - Database schema
- `internal/migrations/020-digest-actions.sql` - Core OHLC actions
- `tests/streams/digest/digest_actions_test.go` - Comprehensive test suite

- Schema migration creates all necessary tables and indexes
- Actions migration implements SQL-compliant OHLC calculation logic
- No breaking changes to existing primitive_events data structure
- Backward compatible with existing time-series queries

resolves: https://github.com/trufnetwork/node/issues/1106

## Summary by CodeRabbit

* **New Features**
  * Added a daily OHLC digestion workflow with batch processing, marker-driven digests, and a read API that falls back to raw data.
  * Improved OPEN/HIGH/LOW/CLOSE determination with explicit tie-break rules and support for combined markers; prunes non-OHLC intraday records.

* **Tests**
  * Added end-to-end tests validating pre/post-digest OHLC, batch digestion, and combination/all-same marker scenarios.

* **Chores**
  * Removed an index related to event stream timing.

### CODE REVIEW

**Comment by MicBun** [comment]
wait

File: internal/migrations/020-digest-actions.sql [hunk 1]
```diff
@@ -1,14 +1,374 @@
...
+    
+    -- Process batch of pending days (oldest first)
+    for $candidate in SELECT stream_ref, day_index FROM pending_prune_days
+                      ORDER BY day_index ASC, stream_ref ASC
+                      LIMIT $batch_size {
+        $deleted INT;
+        $preserved INT;
+        
+        -- Process this day
+        for $result in digest_daily($candidate.stream_ref, $candidate.day_index) {
```
Comment (outerlook) [inline internal/migrations/020-digest-actions.sql:252]: most probably we will need to optimize this process to batch multiple candidates at once (up to a limit of  rows)

because `for...something` are really slow in kwil due to the rountrip

Comment (outerlook) [reply internal/migrations/020-digest-actions.sql:252]: even if it's not for many days at once after we migrate everything, but at least many streams at once... i.e. think about 200K streams emitting just 24 values a day to be updated
